internal/resources/cluster/tkgaws/resource_tkg_aws.go

```diff
@@ -0,0 +1,967 @@
+
+	if v, ok := lookUpRanges[cidrBlocksKey]; ok {
+		ranges.CidrBlocks, _ = v.(string)
+	}
+
+	return ranges
+}
+
+func flattenTKGAWSClusterNetwork(cluster *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsClusterNetwork) (data []interface{}) {
+	flattenClusterNetwork := make(map[string]interface{})
+
+	if cluster == nil {
+		return nil
+	}
+
+	flattenClusterNetwork[apiServerPortKey] = cluster.APIServerPort
+
+	pds := make([]interface{}, 0)
+
+	for _, pd := range cluster.Pods {
+		pds = append(pds, flattenTKGAWSNetworkRange(pd))
+	}
+
+	flattenClusterNetwork[podsKey] = pds
+
+	svs := make([]interface{}, 0)
+
+	for _, sv := range cluster.Services {
+		svs = append(svs, flattenTKGAWSNetworkRange(sv))
+	}
+
+	flattenClusterNetwork[servicesKey] = svs
+
+	return []interface{}{flattenClusterNetwork}
+}
+
+func flattenTKGAWSNetworkRange(ranges *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsNetworkRange) (data interface{}) {
+	flattenRanges := make(map[string]interface{})
+
+	if ranges == nil {
+		return nil
+	}
+
+	flattenRanges[cidrBlocksKey] = ranges.CidrBlocks
+
+	return flattenRanges
+}
+
+var tkgAWSProvider = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Provider Network specifies provider specific network information for the cluster",
+	Required:    true,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			subnetsKey: tkgAWSSubnets,
+			vpcKey:     tkgAWSVPC,
+		},
+	},
+}
+
+func expandTKGAWSProviderNetwork(data []interface{}) (provider *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork) {
+	if len(data) == 0 || data[0] == nil {
+		return provider
+	}
+
+	lookUpProvider, _ := data[0].(map[string]interface{})
+	provider = &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork{
+		Subnets: []*tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsSubnet{},
+		Vpc:     &tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsVPC{},
+	}
+
+	if v, ok := lookUpProvider[subnetsKey]; ok {
+		subnets, _ := v.([]interface{})
+		for _, sn := range subnets {
+			provider.Subnets = append(provider.Subnets, expandTKGAWSSubnets(sn))
+		}
+	}
+
+	if v, ok := lookUpProvider[vpcKey]; ok {
+		if v1, ok := v.([]interface{}); ok {
+			provider.Vpc = expandTKGAWSVPC(v1)
+		}
+	}
+
+	return provider
+}
+
+func flattenTKGAWSProviderNetwork(provider *tkgawsmodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgawsProviderNetwork) (data []interface{}) {
+	flattenProviderNetwork := make(map[string]interface{})
+
+	if provider == nil {
+		return nil
+	}
+
+	sns := make([]interface{}, 0)
+
+	for _, sn := range provider.Subnets {
+		sns = append(sns, flattenTKGAWSSubnets(sn))
+	}
+
+	flattenProviderNetwork[subnetsKey] = sns
+
+	if provider.Vpc != nil {
+		flattenProviderNetwork[vpcKey] = flattenTKGAWSVPC(provider.Vpc)
+	}
+
+	return []interface{}{flattenProviderNetwork}
+}
+
+var tkgAWSSubnets = &schema.Schema{
```

Comment:
Let's also mention the order to be followed for the subnet blocks in the terraform script

Reply:
Where do you exactly want this to be mentioned? README?

Reply:
We can add in the guides as a **NOTE**

Reply:
README seems appropriate

Reply:
guides as a NOTE is also fine, lets capture it